test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -302,6 +302,14 @@ public static int randomInt() {
         return random().nextInt();
     }
 
+    public static long randomPositiveLong() {
+        long positiveLong = randomLong();
```

Comment:
ooh! another wonderful reason to use `do {...} while` syntax! The rarest of syntax!